### ISSUE TITLE
微调模型预测A股的问题

### ISSUE THREAD

**Issue by linyuqi** [shiyu-coder/Kronos#145]
我用A股的数据微调模型，然后用1小时的个股数据作为历史数据参考预测，但是感觉预测的走势太过于平滑，是怎么回事？求教

<img width="2983" height="1483" alt="Image" src="https://github.com/user-attachments/assets/69511030-b2c6-4578-90ad-23cf07f5f6e0" />
参数：T=1.0, top_p=0.95,sample_count=60

**Comment by shiyu-coder** [comment]
我不清楚你这边微调的具体细节。但是如果采样温度T较低，或者sample count比较大，那理论上mean values就是会更加平滑

**Comment by linyuqi** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by linyuqi** [comment]
<img width="2983" height="1483" alt="Image" src="https://github.com/user-attachments/assets/e897b9ee-dc33-40b0-98ea-1b9c12eb2c59" />
我使用的是A股5000多只股票的1H历史数据从2020年1月1日到现在的数据，大概是[TRAIN] Found 25628377 possible samples. Using 100000 per epoch. 2500多万条数据微调的。然后这个图是随便找的一只股票8月22号到今天的小时数据进行预测的，发现历史数据预测背离的情况还是挺严重的，这样我就有点怀疑未来预测的部分，会是什么原因引起的，我的参数默认的T=1.0 temprate=0.9 以下是微调细节，进行到24轮候发现val数值上升就停止学习了
--- Epoch 23/30 Summary ---
Validation Loss: 2.5280
Time This Epoch: 0:15:11
Total Time Elapsed: 5:50:18

[Rank 0, Epoch 24/30, Step 100/2000] LR 0.000005, Loss: 2.4830
[Rank 0, Epoch 24/30, Step 200/2000] LR 0.000005, Loss: 2.5834
[Rank 0, Epoch 24/30, Step 300/2000] LR 0.000005, Loss: 2.5200
[Rank 0, Epoch 24/30, Step 400/2000] LR 0.000005, Loss: 2.4765
[Rank 0, Epoch 24/30, Step 500/2000] LR 0.000005, Loss: 2.4917
[Rank 0, Epoch 24/30, Step 600/2000] LR 0.000005, Loss: 2.4470
[Rank 0, Epoch 24/30, Step 700/2000] LR 0.000005, Loss: 2.4009
[Rank 0, Epoch 24/30, Step 800/2000] LR 0.000005, Loss: 2.5764
[Rank 0, Epoch 24/30, Step 900/2000] LR 0.000005, Loss: 2.5426
[Rank 0, Epoch 24/30, Step 1000/2000] LR 0.000005, Loss: 2.5024
[Rank 0, Epoch 24/30, Step 1100/2000] LR 0.000005, Loss: 2.5424
[Rank 0, Epoch 24/30, Step 1200/2000] LR 0.000005, Loss: 2.4291
[Rank 0, Epoch 24/30, Step 1300/2000] LR 0.000005, Loss: 2.5102
[Rank 0, Epoch 24/30, Step 1400/2000] LR 0.000004, Loss: 2.5535
[Rank 0, Epoch 24/30, Step 1500/2000] LR 0.000004, Loss: 2.5848
[Rank 0, Epoch 24/30, Step 1600/2000] LR 0.000004, Loss: 2.5128
[Rank 0, Epoch 24/30, Step 1700/2000] LR 0.000004, Loss: 2.5401
[Rank 0, Epoch 24/30, Step 1800/2000] LR 0.000004, Loss: 2.4552
[Rank 0, Epoch 24/30, Step 1900/2000] LR 0.000004, Loss: 2.4593
[Rank 0, Epoch 24/30, Step 2000/2000] LR 0.000004, Loss: 2.4724

--- Epoch 24/30 Summary ---
Validation Loss: 2.5268
Time This Epoch: 0:15:12
Total Time Elapsed: 6:05:30

Best model saved to ./outputs/models/finetune_predictor_hour/checkpoints/best_model (Val Loss: 2.5268)

**Comment by HLearning** [comment]
@linyuqi 我和你的做法基本类似， 唯一的区别时， 我训练的时30min的a股票2020以后的数据。loss维持在2.4+， val loss在2.65

**Comment by linyuqi** [comment]
@HLearning 你的预测结果如何？会不会背离实际的太厉害？

**Comment by HLearning** [comment]
> [@HLearning](https://github.com/HLearning) 你的预测结果如何？会不会背离实际的太厉害？

暂时还没有严格的测试 ，等测试后分享出来

**Comment by linyuqi** [comment]
@HLearning 

<img width="2984" height="1483" alt="Image" src="https://github.com/user-attachments/assets/3ef63f65-ea04-4872-8176-677b8ecb6c00" />
你看这结果，我是小时微调，背离的很厉害，这种小盘股，测试感觉预测难度很大，因素太多能决定价格的波动

**Comment by HLearning** [comment]
> [@HLearning](https://github.com/HLearning)
> <img alt="Image" width="2000" height="1483" src="https://private-user-images.githubusercontent.com/11323554/503051277-3ef63f65-ea04-4872-8176-677b8ecb6c00.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA5MzAxMTIsIm5iZiI6MTc2MDkyOTgxMiwicGF0aCI6Ii8xMTMyMzU1NC81MDMwNTEyNzctM2VmNjNmNjUtZWEwNC00ODcyLTgxNzYtNjc3YjhlY2I2YzAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDIwVDAzMTAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5YzQ4ODBkYTk0N2U4NGY1OGExNDEwNTY1Mjg5ZWFmOTRkOWNjNWE3ZmY2NTJiMGEwZjRhZDg0MTU1MjRjMWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.545TumAdQgKmvTtDqHf-ussKJqHJKC3Ae_WkGjfA0vw"> 你看这结果，我是小时微调，背离的很厉害，这种小盘股，测试感觉预测难度很大，因素太多能决定价格的波动

留个wechat， 可以及时交流以下， hpuhjl

**Comment by shiyu-coder** [comment]
这段K线单边下跌，直接预测的话上涨的概率肯定大于继续下跌，所以模型肯定是预测上涨。毕竟只有这么一段单边下跌的K线信息，让人判断也会觉得上涨的概率更大。

**Comment by linyuqi** [comment]
@shiyu-coder 还可以输入更多的参数吗？让结果更优。目前只用了['open', 'close', 'high', 'low', 'volume', 'amount']，这6个参数。

**Comment by lzzh-c912** [comment]
请问一下，A股数据是在哪里下载的呢，我只找到了20年之前的。

**Comment by money0610qj** [comment]
> [@HLearning](https://github.com/HLearning)
> 
> <img alt="Image" width="2000" height="1483" src="https://private-user-images.githubusercontent.com/11323554/503051277-3ef63f65-ea04-4872-8176-677b8ecb6c00.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI4MzE4NTEsIm5iZiI6MTc2MjgzMTU1MSwicGF0aCI6Ii8xMTMyMzU1NC81MDMwNTEyNzctM2VmNjNmNjUtZWEwNC00ODcyLTgxNzYtNjc3YjhlY2I2YzAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTExVDAzMjU1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEwMjkxMThlMWJkMDM0MWViMzhjNjliYzg4YTgyOGEwYWMxOWE3MjVjMWM5YTA0NzZmNDZlODgzYjU2MDM4M2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KgEkqoYGc-WeooApmtVctp0C7V0hlFyIL5WQ-3IEHGs"> 你看这结果，我是小时微调，背离的很厉害，这种小盘股，测试感觉预测难度很大，因素太多能决定价格的波动

请教下这个图是自己做的还是哪块代码生成的

**Comment by world2025** [comment]
@linyuqi 请问下你的训练数据又去掉那些停牌的数据吗，请教下你这边的一个数据处理策略，我是日k线数据，效果不是很好

**Comment by linyuqi** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
